### PR TITLE
ns: Return error when deleting non-existent device

### PR DESCRIPTION
**Pull request by neoaggelos**
#### Summary

References #2323

#### Changes

- Return 404 if deleting a device does not exist

#### Checklist

- [X] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, database and configuration, according to the stability commitments in `README.md`.
- [X] Testing: The changes are covered with unit tests. The changes are tested manually as well.
- [X] Documentation: Relevant documentation is added or updated.
- [X] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [X] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by bafonins** [comment]
works 👌

File: pkg/networkserver/grpc_deviceregistry.go [hunk 1]
```diff
@@ -565,6 +565,9 @@ func (ns *NetworkServer) Delete(ctx context.Context, req *ttnpb.EndDeviceIdentif
 	}
 	var evt events.Event
 	_, _, err := ns.devices.SetByID(ctx, req.ApplicationIdentifiers, req.DeviceID, nil, func(ctx context.Context, dev *ttnpb.EndDevice) (*ttnpb.EndDevice, []string, error) {
+		if dev == nil || !dev.ApplicationIdentifiers.Equal(req.ApplicationIdentifiers) {
```
Comment (rvolosatovs) [inline pkg/networkserver/grpc_deviceregistry.go:568]:
```suggestion
		if dev == nil {
```
https://github.com/TheThingsNetwork/lorawan-stack/pull/2343#discussion_r408112582

Comment (neoaggelos) [reply pkg/networkserver/grpc_deviceregistry.go]: Tried to mirror the behaviour of the Join Server:

https://github.com/TheThingsNetwork/lorawan-stack/blob/0da92e3acedf7e8003b2550c4f3b4d5b7872dacb/pkg/joinserver/grpc_deviceregistry.go#L289

Removed if it does not make sense in the context of the Network Server.

File: CHANGELOG.md [hunk 2]
```diff
@@ -48,6 +48,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Error display on failed end device import in the Console.
 - Update password view not being accessible
 - FOpts encryption and decryption for LoRaWAN 1.1 devices.
+- Network Server returns error when trying to delete a device that does not exist.
```
Comment (rvolosatovs) [inline CHANGELOG.md:51]:
```suggestion
- Network Server returns an error when trying to delete a device that does not exist.
```

File: pkg/networkserver/grpc_deviceregistry.go (hunk 1 above)
Comment (rvolosatovs) [reply pkg/networkserver/grpc_deviceregistry.go]: Yeah, that check makes no sense either anymore it should have been removed in https://github.com/TheThingsNetwork/lorawan-stack/commit/7f5ab3e4fa52db573902af96f147706a7f097e61 when we switched from keying by EUIs to keying by IDs

**Comment by rvolosatovs** [comment]
@neoaggelos can you fix https://github.com/TheThingsNetwork/lorawan-stack/pull/2342#discussion_r409434863 in this PR as well please? Let's make sure everything is consistent and then close https://github.com/TheThingsNetwork/lorawan-stack/issues/2323

**Comment by neoaggelos** [comment]
> @neoaggelos can you fix [#2342 (comment)](https://github.com/TheThingsNetwork/lorawan-stack/pull/2342#discussion_r409434863) in this PR as well please? Let's make sure everything is consistent and then close #2323

Applied changes, can merge once CI finishes

**Comment by rvolosatovs** [comment]
> > @neoaggelos can you fix [#2342 (comment)](https://github.com/TheThingsNetwork/lorawan-stack/pull/2342#discussion_r409434863) in this PR as well please? Let's make sure everything is consistent and then close #2323
> 
> Applied changes, can merge once CI finishes

I don't see the JS change here :thinking:

**Comment by neoaggelos** [comment]
> > > @neoaggelos can you fix [#2342 (comment)](https://github.com/TheThingsNetwork/lorawan-stack/pull/2342#discussion_r409434863) in this PR as well please? Let's make sure everything is consistent and then close #2323
> > 
> > 
> > Applied changes, can merge once CI finishes
> 
> I don't see the JS change here

As shown in #2323, JS behaves properly, and the unit tests verify that you get an error on deleting a non-existing device. https://github.com/TheThingsNetwork/lorawan-stack/blob/4aba5947696a10ec9768365d75e424ea85d643cb/pkg/joinserver/grpc_deviceregistry_test.go#L958

Removing the check for Application Identifiers actually breaks the JS. It seems that the device is not always nil on error. Did not check further why that is the case.

As such, maybe that change is not needed on the JS.

**Comment by rvolosatovs** [comment]
https://github.com/TheThingsNetwork/lorawan-stack/pull/2369

**Pull request merged**